src/runtime/openglcompute.cpp

```diff
@@ -467,17 +471,75 @@ WEAK int halide_openglcompute_run(void *user_context, void *state_ptr,
     // Populate uniforms with values passed in arguments.
     // Order of the passed arguments matches what was generated for this kernel.
     int i = 0;
-    while (arg_sizes[i] != 0) {
+    while (arg_types[i].bits != 0) {
         debug(user_context) << "    args " << i
-                            << " " << (int)arg_sizes[i]
+                            << " " << arg_types[i]
                             << " [" << (*((void **)args[i])) << " ...] "
                             << arg_is_buffer[i] << "\n";
         if (arg_is_buffer[i] == 0) {
-            // TODO(aam): Support types other than int
-            int value = *((int *)args[i]);
-            global_state.Uniform1i(i, value);
-            if (global_state.CheckAndReportError(user_context, "halide_openglcompute_run Uniform1i")) {
-                return -1;
+            if (arg_types[i].code == halide_type_int) {
+                int value;
+                if (arg_types[i].bits == 8) {
+                    value = *((char *)args[i]);
```

Comment:
wouldn't it be preferable to use `int8_t`, `int16_t`, etc here, instead of `int` etc?